scipy/stats/_continuous_distns.py

```diff
@@ -3733,322 +3759,497 @@ class levy_stable_gen(rv_continuous):
-        """Calculate cdf using Zolotarev's methods as detailed in [BS].
+    def _cdf_single_value_piecewise(x, alpha, beta):
+        """Calculate cdf using Nolan's methods as detailed in [NO].
         """
-        zeta = -beta*np.tan(np.pi*alpha/2.)
+        zeta = -beta * np.tan(np.pi * alpha / 2.)
         if alpha != 1:
             x0 = x + zeta  # convert to S_0 parameterization
-            xi = np.arctan(-zeta)/alpha
+            xi = np.arctan(-zeta) / alpha
 
             def V(theta):
-                return np.cos(alpha*xi)**(1/(alpha-1)) * \
-                                (np.cos(theta)/np.sin(alpha*(xi+theta)))**(alpha/(alpha-1)) * \
-                                (np.cos(alpha*xi+(alpha-1)*theta)/np.cos(theta))
+                return np.cos(alpha * xi)**(1 / (alpha - 1)) * (
+                    np.cos(theta) / np.sin(alpha * (xi + theta))
+                )**(alpha / (alpha - 1)) * (
+                    np.cos(alpha * xi + (alpha - 1) * theta) / np.cos(theta)
+                )
+
             if x0 > zeta:
-                c_1 = 1 if alpha > 1 else .5 - xi/np.pi
+                c_1 = 1 if alpha > 1 else .5 - xi / np.pi
 
                 def f(theta):
-                    return np.exp(-V(theta)*np.real(np.complex(x0-zeta)**(alpha/(alpha-1))))
+                    return np.exp(
+                        -V(theta) *
+                        np.real(np.complex(x0 - zeta)**(alpha / (alpha - 1)))
+                    )
 
                 with np.errstate(all="ignore"):
                     # spare calculating integral on null set
                     # use isclose as macos has fp differences
-                    if np.isclose(-xi, np.pi/2, rtol=1e-014, atol=1e-014):
+                    if np.isclose(-xi, np.pi / 2, rtol=1e-014, atol=1e-014):
                         intg = 0
                     else:
-                        intg = integrate.quad(f, -xi, np.pi/2)[0]
-                    return c_1 + np.sign(1-alpha) * intg / np.pi
+                        intg = integrate.quad(f, -xi, np.pi / 2)[0]
+                    return c_1 + np.sign(1 - alpha) * intg / np.pi
             elif x0 == zeta:
-                return .5 - xi/np.pi
+                return .5 - xi / np.pi
             else:
-                return 1 - levy_stable_gen._cdf_single_value_zolotarev(-x, alpha, -beta)
+                return 1 - levy_stable_gen._cdf_single_value_piecewise(
+                    -x, alpha, -beta
+                )
 
         else:
-            # since location zero, no need to reposition x for S_0 parameterization
-            xi = np.pi/2
+            # since location zero, no need to reposition x for S_0
+            # parameterization
+            xi = np.pi / 2
             if beta > 0:
 
                 def V(theta):
-                    expr_1 = np.pi/2+beta*theta
-                    return 2. * expr_1 * np.exp(expr_1*np.tan(theta)/beta) / np.cos(theta) / np.pi
+                    expr_1 = np.pi / 2 + beta * theta
```

Comment:
This function also gets duplicated between `pdf` and `cdf` versions. Again, duplication isn't always so terrible--just bringing this up.